repos/spack_repo/builtin/packages/py_configargparse/package.py

```diff
@@ -19,11 +19,11 @@ class PyConfigargparse(PythonPackage):
     features."""
 
     homepage = "https://github.com/bw2/ConfigArgParse"
-    url = "https://github.com/bw2/ConfigArgParse/archive/1.7.tar.gz"
+    url = "https://github.com/bw2/ConfigArgParse/archive/1.7.1.tar.gz"
 
     license("MIT")
 
-    version("1.7", sha256="4549d105790386d01f71beebc3aa457d4177315680b75415f05bc22e1e28183a")
+    version("1.7.1", sha256="b31bfcce183ef7226f00cd07a37cfd69fe654229fe462f9a1de6091f396fc0f9")
```

Comment:
I suspect the Spack solver won’t be too pleased if the version simply disappears, with package being already installed. Removing versions this way is against Spack policy.

Consider bringing back `1.7` marked as `deprecated=True`

```
version("1.7", sha256="4549d105790386d01f71beebc3aa457d4177315680b75415f05bc22e1e28183a", deprecated=True)
```

Reply:
Ok, then you know Spack internals better than I do. I just thought it is a strict rule to not remove hashes without deprecation, whatever the reason.

It can be merged then.